tidy3d/plugins/dispersion/fit_fast.py

```diff
@@ -0,0 +1,834 @@
+
+        Note
+        ----
+        The algorithm is described in::
+
+            B. Gustavsen and A. Semlyen, "Rational approximation
+            of frequency domain responses by vector fitting,"
+            IEEE Trans. Power. Deliv. 14, 3 (1999).
+
+            B. Gustavsen, "Improving the pole relocation properties
+            of vector fitting," IEEE Trans. Power Deliv. 21, 3 (2006).
+
+            B. Gustavsen, "Enforcing Passivity for Admittance Matrices
+            Approximated by Rational Functions," IEEE Trans. Power
+            Syst. 16, 1 (2001).
+
+        Note
+        ----
+        The fit is performed after weighting the real and imaginary parts,
+        so the RMS error is also weighted accordingly. By default, the weights
+        are chosen based on typical values of the data. To change this behavior,
+        use 'AdvancedFastFitterParam.weights'.
+
+
+        Parameters
+        ----------
+        max_num_poles: PositiveInt, optional
+            Maximum number of poles in the model.
+        eps_inf : float, optional
+            Value of eps_inf to use in fit. If None, then eps_inf is also fit.
+            Note: fitting eps_inf is not guaranteed to yield a global optimum, so
+            the result may occasionally be better with a fixed value of eps_inf.
+        tolerance_rms : float, optional
+            Weighted RMS error below which the fit is successful and the result is returned.
+        advanced_param : :class:`AdvancedFastFitterParam`, optional
+            Advanced parameters for fitting.
+
+        Returns
+        -------
+        Tuple[:class:`.PoleResidue`, float]
+            Best fitting result: (dispersive medium, weighted RMS error).
+        """
+
+        omega = 2 * np.pi * HBAR * self.freqs[::-1]
+        eps = self.eps_data[::-1]
+
+        init_model = FastFitterData.initialize(
+            omega, eps, eps_inf, advanced_param or AdvancedFastFitterParam()
+        )
+        if init_model.verbose:
+            log.info(
+                f"Fitter: initializing, weights=({init_model.weights[0]:.3g}, "
+                f"{init_model.weights[1]:.3g})"
+            )
+
+        def make_configs():
+            configs = [[p] for p in range(max(init_model.min_num_poles // 2, 1), max_num_poles + 1)]
+            for setting in [
+                init_model.relaxed,
+                init_model.smooth,
+                init_model.logspacing,
+                init_model.optimize_eps_inf,
+            ]:
+                if setting is None:
+                    configs = [c + [r] for c in configs for r in [True, False]]
+                else:
+                    configs = [c + [r] for c in configs for r in [setting]]
+            return configs
+
+        best_model = init_model
+        warned_about_passivity_num_iters = False
+        warned_about_slsqp_constraint_scale = False
+
+        configs = make_configs()
+
+        with Progress() as progress:
```

Comment:
we might only want progress bar if verbose, is it possible to set it up this way?